modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -443,18 +455,45 @@ private CompletableFuture<Void> createZonePartitionReplicationNode(
 
         Supplier<CompletableFuture<Boolean>> startReplicaSupplier = () -> {
             try {
-                replicationGroupIds.add(replicaGrpId);
+                CatalogZoneDescriptor zoneDescriptor = catalogMgr.zone(replicaGrpId.zoneId(), catalogMgr.latestCatalogVersion());
+
+                AtomicReference<Long> stamp = new AtomicReference<>(null);
 
                 return replicaMgr.startReplica(
-                        replicaGrpId,
-                        (raftClient) -> new ZonePartitionReplicaListener(
-                                new ExecutorInclinedRaftCommandRunner(raftClient, partitionOperationsExecutor)),
-                        new FailFastSnapshotStorageFactory(),
-                        stablePeersAndLearners,
-                        raftGroupListener,
-                        raftGroupEventsListener,
-                        busyLock
-                ).thenApply(ignored -> true);
+                                replicaGrpId,
+                                (raftClient) -> new ZonePartitionReplicaListener(
+                                        new ExecutorInclinedRaftCommandRunner(raftClient, partitionOperationsExecutor)),
+                                new FailFastSnapshotStorageFactory(),
+                                stablePeersAndLearners,
+                                raftGroupListener,
+                                raftGroupEventsListener,
+                                busyLock
+                        ).thenCompose(replica -> {
+                            zonePartitionsLocks.compute(zoneId, (id, lock) -> {
+                                if (lock == null) {
+                                    lock = new StampedLock();
+                                }
+
+                                stamp.set(lock.writeLock());
+
+                                return lock;
+                            });
+
+                            replicationGroupIds.add(replicaGrpId);
+
+                            return fireEvent(
+                                    LocalPartitionReplicaEvent.AFTER_REPLICA_STARTED,
+                                    new PartitionReplicaEventParameters(
+                                            new ZonePartitionId(zoneDescriptor.id(), replicaGrpId.partitionId())
```

Comment:
why do we need to call 
```
CatalogZoneDescriptor zoneDescriptor = catalogMgr.zone(replicaGrpId.zoneId(), catalogMgr.latestCatalogVersion());
```
if we already have zoneId passed to `PartitionReplicaLifecycleManager#createZonePartitionReplicationNode?`

Please, do not use construction like 

```
CatalogZoneDescriptor zoneDescriptor = catalogMgr.zone(replicaGrpId.zoneId(), catalogMgr.latestCatalogVersion());
```
 it could be unsafe and zone could be already removed

Reply:
We had disccussions why it will valid in this case, but it doesn't matter, you are right, I don't need full descriptor at all

Reply:
Fixed